doc/man3/SSL_get_value_uint.pod

```diff
@@ -162,18 +178,82 @@ of QUIC stream creation flow control.
+
+If set to this value (the default), the implicit event handling model is used.
+Under this model, QUIC objects will automatically perform background event
+processing (equivalent to a call to L<SSL_handle_events(3)>) when calls to I/O
+functions such as L<SSL_read_ex(3)> or L<SSL_write_ex(3)> are made on a QUIC SSL
+object. This helps to maintain the health of the QUIC connection and ensures
+that incoming datagrams and timeout events are processed.
+
+=item B<SSL_VALUE_EVENT_HANDLING_MODE_EXPLICIT> (Explicit event handling)
+
+If set to 0, the explicit event handling model is used. Under this model,
+B<nonblocking> calls to I/O functions such as L<SSL_read_ex(3)> or
+L<SSL_write_ex(3)> do not result in the automatic processing of QUIC events. Any
+new incoming network traffic is not handled; no new outgoing network traffic is
+generated, and pending timeout events are not processed. This allows an
+application to obtain greater control over the circumstances in which QUIC event
+processing occurs. If this event handling model is used, it is the application's
+responsibility to call L<SSL_handle_events(3)> as and when called for by the
+QUIC implementation; see the L<SSL_get_rpoll_descriptor(3)> man page for more
+information.
+
+Selecting this model does not affect the operation of blocking I/O calls, which
+will continue to use the implicit event handling model. Therefore, applications
+using this model will generally want to disable blocking operation using
+L<SSL_set_blocking_mode(3)>.
+
+=back
+
+Can be configured using the convenience macros SSL_get_event_handling_mode() and
+SSL_set_event_handling_mode().
+
+A call to SSL_set_value_uint() which causes this value to switch back to the
+implicit event handling model does not in itself cause implicit event handling
+to occur; such handling will occur on the next I/O API call. Equally, a call to
+SSL_set_value_uint() which causes this value to switch to the explicit event
+handling model will not cause event handling to occur before making that
+transition.
+
+This value controls whether implicit event handling occurs when making an I/O
+API call on the SSL object it is set on. However, event processing is not
+confined to state which relates to only that object. For example, if you
+configure explicit event handling on QUIC stream SSL object "A" and configure
+implicit event handling on QUIC stream SSL object "B", a call to an I/O function
+on "B" may result in state changes to "A". In other words, if event handling
+does happen as a result of an API call to an object related to a connection,
+processing of background events (for example, received QUIC network traffic) may
+also affect the state of any other object related to a connection.
+
 =back
 
-No configurable values are currently defined for QUIC stream SSL objects or for
-other kinds of SSL object.
+No configurable values are currently defined for non-QUIC SSL objects.
 
 =head1 RETURN VALUES
 
```

Comment:
I'm quite confused by the difference between SSL_VALUE_EVENT_HANDLING_MODE_IMPLICIT and SSL_VALUE_EVENT_HANDLING_MODE_INHERIT.

If a connection SSL object is set to SSL_VALUE_EVENT_HANDLING_MODE_IMPLICIT, and then a new stream is created - what value does the stream use? How is that different to if SSL_VALUE_EVENT_HANDLING_MODE_INHERIT is set?

Similarly if a connection SSL object is set to SSL_VALUE_EVENT_HANDLING_MODE_EXPLICIT and a new stream is then created - what value does the stream use?

Does setting the value on a connection SSL object impact stream objects that are already in existence?

Reply:
A new stream object defaults to `INHERIT`. So it defaults to whatever is currently set on the connection object. This differs from a design in which the current value at stream creation time is simply copied from the connection object, as it means future changes to the value on the connection object also affect stream objects unless specifically overridden.